File: dotnet/src/Microsoft.Agents.Validation/Validators/ContentValidators/DocumentContentValidator.cs
using Microsoft.Agents.Xml.Generated.Models;

namespace Microsoft.Agents.Xml.Validation.Validators.ContentValidators;

/// <summary>
/// Validator for DocumentContent.
/// </summary>
public class DocumentContentValidator : ContentValidatorBase<DocumentContent>
{
    public override ValidationResult Validate(DocumentContent content, ValidationContext? context = null)
    {
        var errors = new List<ValidationError>();

        // CNT-013: DocumentContent document-id must be non-empty
        var docIdError = ValidateNotEmpty(content.DocumentId, "DocumentId",
            ValidationErrorCode.CNT_013,
            "DocumentContent document-id must be non-empty");
        if (docIdError != null)
            errors.Add(docIdError);

        return new ValidationResult(errors);
    }
}

File: typescript/packages/agents-protocol-validation/src/validators/content/DocumentContentValidator.ts
import type { DocumentContent } from '@microsoft/agents-protocol-abstractions';
import type { ValidationContext } from '../../core/ValidationContext';
import { ContentValidatorBase } from '../base/ContentValidatorBase';
import { ValidationResult } from '../../core/ValidationResult';

/**
 * Validator for DocumentContent.
 */
export class DocumentContentValidator extends ContentValidatorBase<DocumentContent> {
  public validate(content: DocumentContent, context?: ValidationContext): ValidationResult {
    // DocumentContent has no strict validation requirements
    return ValidationResult.success();
  }
}
